### PR TITLE
[docs] Reshuffle columns and rows styling sections

### LINKED ISSUE #1496: [docs] Update the style page to illustrate how to style rows, cell and column headers
We want to centralize how to style the datagrid so users can decide which approach they need.

In columns
- Move sections
https://material-ui.com/components/data-grid/columns/#styling-header
https://material-ui.com/components/data-grid/columns/#styling-cells
- add a link to style page corresponding section

Add a section to show how to style rows using 'getRowClassname'

### PR DESCRIPTION
**Pull request by DanailH**
Fixes #1496 

Sections already existed so this was more of combining them in a single page.

Things I'm not sure about:
- The right side nav structure - for `columns` we have 2 separate sections but for `rows` there is only one,
- Do we still need the demo? I've moved it to the bottom of the page. Technically it is a duplicate because the same can be found on the Demo page
- Should update the examples?

https://deploy-preview-1622--material-ui-x.netlify.app/components/data-grid/style/

### CODE REVIEW

**Comment by oliviertassinari** [comment]
> Do we still need the demo? I've moved it to the bottom of the page. Technically it is a duplicate because the same can be found on the Demo page

I think so because the purpose is different. In the one from the /demo page, developers are not supposed to look at the source it will grow as a mess as we make it full-featured.

**Comment by DanailH** [comment]
I agree, the demo feels off as we already have a demo on another page. I would change it to "Customized Demo" (single) and if we add more we can have sections and rename it to "Customized Demos"

File: docs/src/pages/components/data-grid/style/style.md [hunk 1]
```diff
@@ -7,6 +7,99 @@ components: DataGrid, XGrid
...
+        negative: (params.value as number) < 0,
+        positive: (params.value as number) > 0,
+      }),
+  },
+];
+```
+
+{{"demo": "pages/components/data-grid/style/StylingCellsGrid.js", "bg": "inline"}}
+
+## Customized demo
```
Comment (dtassone) [inline docs/src/pages/components/data-grid/style/style.md:101]: `Css customization`?

As it's bit more specific...

Comment (oliviertassinari) [reply docs/src/pages/components/data-grid/style/style.md:101]: The page is already about CSS customizations 😁.

```suggestion
## Custom theme
```

?

Comment (dtassone) [reply docs/src/pages/components/data-grid/style/style.md:101]: Well the page is about styling, so why not 😄 

Anyway, I like the idea of mentioning `theme` in the title. 
`Theming with CSS` so so 
`Css Theme`  or 
`Custom Theme` or just `Theme`?

Comment (DanailH) [reply docs/src/pages/components/data-grid/style/style.md:101]: Honestly `theme` sounds a bit weird (but so does `demo`). We can always change it if we don't feel like it makes sense in the future

**Review by dtassone — APPROVED** [review]

**Pull request merged**